docs/content/how-to/aws/create-aws-hosted-cluster-autonode.md

```diff
@@ -0,0 +1,255 @@
+
+The following steps describes how to install OpenShift workload cluster
+on AWS with AutoNode feature by Karpenter.
+
+[ec2nodeclass]: https://karpenter.sh/docs/concepts/nodeclasses/
+[nodepools]: https://karpenter.sh/docs/concepts/nodepools/
+
+## Prerequisites
+
+- [Install the latest and greatest HyperShift CLI](https://hypershift-docs.netlify.app/getting-started/#prerequisites).
+    - Make sure all prerequisites have been satisfied (Pull Secret, Hosted Zone, OIDC Bucket, etc)
+- Enable AWS service-linked role for spot (one time by account where the hosted cluster will be installed):
+```sh
+aws iam create-service-linked-role --aws-service-name spot.amazonaws.com
+```
+
+### Create Karpenter IAM Role
+
+Create the IAM Role used by Karpenter controller.
+
+!!! warning "Temporary Step"
+    This is a temporary step while the issue [PODAUTO-302][PODAUTO-302]
+    is delivered.
+
+[PODAUTO-302]: https://issues.redhat.com/browse/PODAUTO-302
+
+```sh
+export ROLE_NAME=KarpenterNodeRole-agl
+aws iam create-role --role-name $ROLE_NAME --assume-role-policy-document '{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ec2.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}'
+
+aws iam attach-role-policy --role-name $ROLE_NAME --policy-arn arn:aws:iam::aws:policy/AdministratorAccess
+```
+
+### Export environment variables
+
+Export the following environment variables, adjusting according to your environment:
+
+```sh
+# AWS config
+export AWS_CREDS="$HOME/.aws/credentials"
+export AWS_DEFAULT_REGION=us-east-1
+
+# OpenShift credentials and configuration
+export CLUSTER_PREFIX=hcp-aws
+export CLUSTER_BASE_DOMAIN=devcluster.openshift.com
+export PULL_SECRET_FILE="${HOME}/.openshift/pull-secret-latest.json"
+export SSH_PUB_KEY_FILE=$HOME/.ssh/id_rsa.pub
+
+# Hypershift configuration
+## S3 Bucket name hosting the OIDC discovery documents
+export OIDC_BUCKET_NAME="${CLUSTER_PREFIX}-oidc"
+```
+
+### Install the Hypershift Operator
+
+Install controller enabling `tech-preview-no-upgrade` feature gate:
+
+```sh
+./hypershift install \
+  --oidc-storage-provider-s3-bucket-name="${OIDC_BUCKET_NAME}" \
+  --oidc-storage-provider-s3-credentials="${AWS_CREDS}" \
+  --oidc-storage-provider-s3-region="${AWS_DEFAULT_REGION}" \
+  --tech-preview-no-upgrade=true
+```
+
+- Check if controller is running as expected:
+
+```sh
+oc get all -n hypershift
+```
+
+## Create Workload Cluster with AutoNode
+
+Create the workload cluster with HyperShift AutoNode.
+
+Choose the desired target release image name ([release controller](https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/)).
+
+Create a hosted cluster, enabling the flag `--auto-node`:
+
+```sh
+HOSTED_CLUSTER_NAME=${CLUSTER_PREFIX}-419-ng
+OCP_RELEASE_IMAGE=registry.ci.openshift.org/ocp/release:4.19.0-0.nightly-2025-01-21-163021
```

Comment:
Maybe we update this to an updated release image with all the most recent karpenter/autonode changes such as: `quay.io/openshift-release-dev/ocp-release:4.19.0-rc.5-x86_64`

Or put in a dummy image like
`OCP_RELEASE_IMAGE=LATEST_RELEASE_IMAGE`

So that people don't look at the tutorial and just use that old release image for the guest cluster without thinking.